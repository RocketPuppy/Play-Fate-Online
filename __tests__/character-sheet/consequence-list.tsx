import * as React from "react";
import * as renderer from "react-test-renderer";
import ConsequenceList from "../../src/character-sheet/consequence-list";

const noop = () => {};

test("renders nothing with no consequences", () => {
  const render = renderer.create(<ConsequenceList consequences={[]} onChange={noop} />);

  expect(render.toJSON()).toMatchSnapshot();
});

test("renders a single consequences", () => {
  const consequences = [
    { description: "One", level: 1 }
  ];
  const render = renderer.create(<ConsequenceList consequences={consequences} onChange={noop} />);

  expect(render.toJSON()).toMatchSnapshot();
});

test("renders some consequences", () => {
  const consequences = [
    { description: "One", level: 1 },
    { description: "Two", level: 2 },
    { description: "Three", level: 3 }
  ];
  const render = renderer.create(<ConsequenceList consequences={consequences} onChange={noop} />);

  expect(render.toJSON()).toMatchSnapshot();
});
