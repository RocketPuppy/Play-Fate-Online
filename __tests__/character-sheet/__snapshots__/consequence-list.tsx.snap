// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single consequences 1`] = `
<div>
  <label>
    <span>
      1
    </span>
    <span />
    <input
      defaultValue="One"
      onChange={[Function]}
      type="text"
    />
  </label>
</div>
`;

exports[`renders nothing with no consequences 1`] = `<div />`;

exports[`renders some consequences 1`] = `
<div>
  <label>
    <span>
      1
    </span>
    <span />
    <input
      defaultValue="One"
      onChange={[Function]}
      type="text"
    />
  </label>
  <label>
    <span>
      2
    </span>
    <span>
      Mild
    </span>
    <input
      defaultValue="Two"
      onChange={[Function]}
      type="text"
    />
  </label>
  <label>
    <span>
      3
    </span>
    <span />
    <input
      defaultValue="Three"
      onChange={[Function]}
      type="text"
    />
  </label>
</div>
`;
